### DISCUSSION TITLE
Scopes des Conventional Commits

### DISCUSSION THREAD

**Discussion by seba1204** [photonsquid/7map#27] [Ideas]
Avec l'extension [Conventional Commits](https://marketplace.visualstudio.com/items?itemName=vivaxy.vscode-conventional-commits) de VS Code, il est possible de définir des scopes dans le fichier de paramétrage du Workspace  (.vscode/settings.json) :
```
"conventionalCommits.scopes": ["data", "rendering"]
```
Je trouve qu'il serait pertinent de partager ces infos avec toute l'équipe, pour éviter que l'on aie un scope `data`, un `Data`, un `données`, `donnée`, `Données`, ... 
Mais quelqu'un m'a dit que mettre le fichier `settings.json` dans un repos public n'était pas une best-practice, puisque chacun a une configuration différente. 

Des idées pour arranger ça ?

**Comment by BSoDium** [comment]
à part s'organiser correctement pour que tout le monde crée des scopes personnels qui vérifient les conventions, je vois pas personnellement, sachant que tout le monde n'est pas sur vs code, c'est pas vraiment possible de l'automatiser à mon avis 😕

**Reply by seba1204** [reply]
Selong la [documetation officielle de VS Code ](https://code.visualstudio.com/docs/getstarted/settings): 
"Workspace settings override user settings. Workspace settings are specific to a project and can be shared across developers on a project."

J'ai déjà vu des "boilerplate" avec un settings.jons il me semble, contenant une propritété du style "recommanded packages", avec une liste d'extensions, et quand tu ouvres le workspace, VS Code te propose de les installer si tu ne les a pas déjà.
Je trouve que cet exemple montre bien qu'on a le droit de partager les paramètres du Workspace 😄

**Reply by seba1204** [reply]
> sachant que tout le monde n'est pas sur vs code

L'idée était bien de partager les paramètres de l'extension, donc uniquement pour ceux qui ont VS Code 🤷

**Reply by BSoDium** [reply]
dans ce cas fais toi plaisir j'ai envie de dire, crée un workspace et commit le, tant qu'il n'y a pas de chemin absolus qui trainent dedans et qui pourraient engendrer des conflits chez les autres ça devrait être ok 👌
